test/unit/spec/EnrollChoices_spec.js

```diff
@@ -689,232 +716,319 @@ function (Okta, OktaAuth, Util, EnrollChoicesForm, Beacon, Expect, FactorUtil, R
+            //optional factor should have button and no cardinality text
+            expect(test.form.factorButtonText(null, optionalFactorRow)).toBe('Set up another');
+            expect(test.form.factorCardinalityText(null, optionalFactorRow)).toBe('');
+          });
         });
         itp('displays correct cardinality text when enrolled=3 required=0 optional=0', function () {
-          return setupMultipleFactorEnrollments(factorName, 'ACTIVE', 'REQUIRED',
-            {'enrolled': 3, 'minimum': 2, 'maximum': 3})
-            .then(function (test) {
-              var factorRows = test.form.factorRow(factorName);
-              //displays only one row for enrolled, none for optional
-              expect(factorRows.length).toBe(1);
-              //enrolled factor row should have success check and cardinality text
-              expect(test.form.factorHasSuccessCheck(factorName)).toBe(true);
-              expect(test.form.factorCardinalityText(factorName)).toBe('(3 set up)');
-            });
+          return setupMultipleFactorEnrollments({
+            factorName: factorName,
+            status: 'ACTIVE',
+            enrollment: 'REQUIRED',
+            policyEnrollment: {'enrolled': 3, 'minimum': 2, 'maximum': 3},
+            webauthnEnabled: webauthnEnabled,
+            useProfiles: useProfiles
+          }).then(function (test) {
+            var factorRows = test.form.factorRow(factorName);
+            //displays only one row for enrolled, none for optional
+            expect(factorRows.length).toBe(1);
+            //enrolled factor row should have success check and cardinality text
+            expect(test.form.factorHasSuccessCheck(factorName)).toBe(true);
+            expect(test.form.factorCardinalityText(factorName)).toBe('(3 set up)');
+          });
         });
         itp('checks that the order of factors is correct during enroll if enrollment OPTIONAL', function () {
-          return setupMultipleFactorEnrollments(
-            factorName, 'ACTIVE', 'OPTIONAL',{'enrolled': 1, 'minimum': 0, 'maximum': 3}
-          ).then(function (test) {
+          return setupMultipleFactorEnrollments({
+            factorName: factorName,
+            status: 'ACTIVE',
+            enrollment: 'OPTIONAL',
+            policyEnrollment: {'enrolled': 1, 'minimum': 0, 'maximum': 3},
+            webauthnEnabled: webauthnEnabled,
+            useProfiles: useProfiles
+          }).then(function (test) {
             var factorRows = test.form.optionalFactorList();
             var factorIndex = factorRows.find(`[data-se="${factorName}"]`).index();
-            var expectedIndex = FactorUtil.getFactorSortOrder(factorEnrollList[factorName].provider, factorEnrollList[factorName].factorType) - 1;
+            var expectedIndex = FactorUtil.getFactorSortOrder.call(test.router, factorEnrollList[factorName].provider, factorEnrollList[factorName].factorType) - 1;
```

Comment:
don't see why this changed was needed. It's good, just curious what I missed

Reply:
It fails only for webauthn, because it calls `FactorUtil.getFactorNameForFactorType` and that references the `settings` object to get `features.webauthn`. So i had to set the settings object in router (in set up above) and pass it as `this` when i call this function.